app/merchant/views.py

```diff
@@ -14,31 +14,52 @@ def index():
 
 
 @merchant.route('/view/all')
+@merchant.route('/view/all/<int:page>')
 @login_required
 @merchant_required
-def listing_view_all():
+def listing_view_all(page=1):
     """Search for listings"""
-    main_search_term = request.args.get('main-search', "", type=str)
+    main_search_term = request.args.get('mainsearch', "", type=str)
```

Comment:
Why is the query parameter no longer dashed? It's a bit harder to read as a single word